src/autoPlay.js

```diff
@@ -102,12 +102,17 @@ export default function autoPlay(MyComponent) {
 
     handleInterval = () => {
       const {
+        autoplay,
         children,
         direction,
         onChangeIndex,
         slideCount,
       } = this.props;
 
+      if (!autoplay) {
+        clearInterval(this.timer);
```

Comment:
I thought we could reach this state. How do you end up here? Sounldn't the React  life cycle hook prevent it in the first place?